### ISSUE TITLE
busy loop with `make install`

### ISSUE THREAD

**Issue by eworm-de** [acassen/keepalived#1964]
**Describe the bug**
Configuring with:
```
  ./configure \
    --prefix=/usr \
    --sysconfdir=/etc \
    --sbindir=/usr/bin \
    --localstatedir=/var \
    --runstatedir=/run
```
Building (`make`) succeeds, but installing (`make DESTDIR="$pkgdir" install`) results in an infinite loop.

Looks like this happens in `make install-exec-hook` and was introduced with 81b5d68d08b2661f3256e0a673a4627417aaab85.

**To Reproduce**
Configure, build and install.

**Expected behavior**
Should install.

**Keepalived version**
2.2.3

**Distro (please complete the following information):**
 - Name: Arch
 - Version: rolling
 - Architecture: x86_64

**Comment by pqarmitage** [comment]
@eworm-de The actual cause of the problem you experienced was sbindir being the same as bindir. I had already produced a patch to resolve that, but I like your idea of using realpath. Unfortunately the busybox version of realpath does not support `--relative-to`, so we cannot rely on using realpath.

Since I like the idea of using `realpath --relative-to` where possible, I have updated keepalived/Makefile.am to test if it can be use, and if so it does so, otherwise it creates the path the old way, but with a check to make sure bindir and sbindir are different.

Many thanks for you patch, which although it won't be directly merged, the improvement has been incorporated for use when it can.

Commits 76f74d4 and 8de0a3e resolve this issue.
